### PR TITLE
test: create square construction & deconstruction parity test

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: https://github.com/celestiaorg/celestia-app/issues/1729

### CODE REVIEW

**Review by evan-forbes — DISMISSED** [review]

File: pkg/square/square_test.go [hunk 1]
```diff
@@ -366,6 +369,33 @@ func TestSquareBlobShareRange(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestSquareDeconstruct(t *testing.T) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	t.Run("ConstructDecostructParity", func(t *testing.T) {
+		const numTxs = 10
+		txs := generateOrderedTxs(numTxs, numTxs, 100)
```
[the remark this answers is not in the document]
Comment (cmwaters) [reply pkg/square/square_test.go:376]: There are existing tests that generate enough transactions to exceed the size of the square. Do those count?

File: pkg/shares/namespace.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package shares
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/namespace"
+)
+
+// GetSharesByNamespace returns all shares that belong to a given namespace.
```
Comment (rootulp) [inline pkg/shares/namespace.go:9]:
```suggestion
// GetShareRangeForNamespace returns all shares that belong to a given namespace.
```

File: pkg/square/square.go [hunk 3]
```diff
@@ -110,6 +217,19 @@ func (s Square) Equals(other Square) bool {
 	return true
 }
 
+// WrappedPFBs returns the wrapped PFBs in a square
+func (s Square) WrappedPFBs() (core.Txs, error) {
+	wpfbShareRange, err := shares.GetShareRangeForNamespace(s, namespace.PayForBlobNamespace)
+	if err != nil {
+		return core.Txs{}, nil
+	}
+	return shares.ParseTxs(s[wpfbShareRange.Start : wpfbShareRange.End+1])
```
Comment (rootulp) [inline pkg/square/square.go:226]: [question]

```go
// namespace. Ranges here are always end exlusive.
func GetShareRangeForNamespace(shares []Share, ns namespace.Namespace) (Range, error) {
```

so they is the `+1` necessary for `wpfbShareRange.End+1`?

**Review by rootulp — APPROVED** [review]
LGTM w/ one question

File: pkg/square/square.go (hunk 3 above)
Comment (cmwaters) [reply pkg/square/square.go:226]: Good catch! wasn't even looking there

**Review by evan-forbes — APPROVED** [review]
:+1: thanks for adding more tests :slightly_smiling_face:

File: pkg/square/square_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (evan-forbes) [reply pkg/square/square_test.go]: the added tests get there I think!

**Review by rootulp — APPROVED** [review]

**Pull request merged**